scripts/autotools/libmongoc/CheckUtf8Proc.m4

```diff
@@ -0,0 +1,40 @@
+PHP_ARG_WITH([mongodb-utf8proc],
+             [whether to use system or bundled utf8proc for SCRAM-SHA-256 SASLprep],
+             [AS_HELP_STRING([--with-mongodb-utf8proc=@<:@bundled/system@:>@],
+                             [MongoDB: Enable utf8proc for SCRAM-SHA-256 SASLprep [default=bundled]])],
+             [bundled],
+             [no])
+
+found_utf8proc="no"
+bundled_utf8proc="no"
+
+AS_IF([test "$PHP_MONGODB_UTF8PROC" = "system"],[
+  PKG_CHECK_MODULES([PHP_MONGODB_UTF8PROC],[libutf8proc],[
+    PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS $PHP_MONGODB_UTF8PROC_CFLAGS"
+    PHP_EVAL_LIBLINE([$PHP_MONGODB_UTF8PROC_LIBS],[MONGODB_SHARED_LIBADD])
+    found_utf8proc="yes"
+  ],[
+    PHP_CHECK_LIBRARY([utf8proc],
+                      [utf8proc_NFKC],
+                      [have_utf8proc_lib="yes"],
+                      [have_utf8proc_lib="no"])
+
+    AC_CHECK_HEADER([utf8proc.h],
+                    [have_utf8proc_headers=yes],
+                    [have_utf8proc_headers=no])
+
+    if test "$have_utf8proc_lib" = "yes" -a "$have_utf8proc_headers" = "yes"; then
+      PHP_ADD_LIBRARY([utf8proc],,[MONGODB_SHARED_LIBADD])
+      found_utf8proc="yes"
+    fi
```

Comment:
Note: I tested both of these code paths locally.

```
$ pkg-config  libutf8proc --libs
-lutf8proc
```

```
pkg-config  libutf8proc --cflags
-DUTF8PROC_EXPORTS
```

Configure only shows `-lutf8proc`, but I do see `-DUTF8PROC_EXPORTS` added to `$PHP_MONGODB_BUNDLED_CFLAGS` when using pkg-config. I'm not sure if it's required for the second branch. /cc @joshbsiegel

Reply:
I don't think `UTF8PROC_EXPORTS` is relevant, as it's only referenced in the following conditional for Windows:

https://github.com/JuliaStrings/utf8proc/blob/1cb28a66ca79a0845e99433fd1056257456cef8b/utf8proc.h#L123-L137

Reply:
I agree. I didn't have to do anything with `UTF8PROC_EXPORTS` when making changes in the C driver.